crates/uv-cli/src/lib.rs

```diff
@@ -4908,6 +4918,74 @@ pub struct PythonInstallArgs {
     pub default: bool,
 }
 
+#[derive(Args)]
+#[allow(clippy::struct_excessive_bools)]
+pub struct PythonUpgradeArgs {
+    /// The directory to store the Python installation in.
+    ///
+    /// If provided, `UV_PYTHON_INSTALL_DIR` will need to be set for subsequent operations for uv to
+    /// discover the Python installation.
+    ///
+    /// See `uv python dir` to view the current Python installation directory. Defaults to
+    /// `~/.local/share/uv/python`.
+    #[arg(long, short, env = EnvVars::UV_PYTHON_INSTALL_DIR)]
+    pub install_dir: Option<PathBuf>,
+
+    /// The Python minor version(s) to upgrade.
+    ///
+    /// If not provided, uv will attempt to upgrade every minor version that has
+    /// been installed.
+    #[arg(env = EnvVars::UV_PYTHON)]
+    pub targets: Vec<String>,
```

Comment:
How will this interface work in the future if / when we add support for upgrading other implementations, like PyPy?

Reply:
It's definitely supported in the `PythonRequest` syntax / seems fairly reasonable to me.